### DISCUSSION TITLE
Cancel pipeline as soon as a newer version comes in

### DISCUSSION THREAD

**Discussion by drahnr** [concourse/concourse#6764] [Help & Support] [answered]
I have a rather expensive CI pipeline for PRs, currently they are configured as `serial: true` to avoid OOM conditions, but that triggers other issues for fast moving PRs, since the first execution might not even have finished by the time a new commit is pushed to the pipeline.

Is there a way to cancel the previous pipeline run when a new one comes in?

**Comment by cynicaljoy** [comment]
for reference, looking to achieve https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency `cancel-in-progress`

**Reply by KoltesDigital** [reply]
Drone also has this feature, to be found in the settings of each active repository.

> Automatically cancel running builds if newer commit pushed.

**Comment by marco-m** [comment]
I don't think so. Assuming that you are using the Concourse PR resource, it might be possible to modify it to issue a query to the ATC and send an abort command (same as `fly abort-build`) to the running builds... a bit of an hack...

Pointing out the obvious just to cover all possibilities: it it also always possible for the user to issue `fly abort-build` directly, just after having pushed another version, or to use the Concourse UI to do the same.

**Comment by cynicaljoy** [comment]
Yeah, tentative plan right now is to write a `task` that uses the [fly resource](https://github.com/troykinsella/concourse-fly-resource) to essentially invoke:

```
fly builds -p $PIPELINE --json | jq -r '[.[] | select(.status | IN("started", "pending")).id] | del(.[0]) | .[]' | xargs -L1 -I {} fly abort-build -b {}
```

**Comment by vlfig-iov42** [comment]
Presumably it wouldn't be against the grain to honour the revisions as reported by the resource, and thus cancel whichever builds depend on a revision that no longer exists.

It would then behoove a resource such as the PR one (or the git resource itself) to not report orphaned commits as revisions.

Just thinking out loud here. Maybe there's some gotcha to this.

**Comment by taylorsilva** [comment, marked as answer]
Just came across this through a notification rabbit hole.
For anyone wondering, this currently isn't a feature in Concourse. I think this would be a great thing to add. Anyone is free to take this on and make a PR implementing this. Probably some kind of field on the job key, at the same level as `plan`.

**Reply by chuxubank** [reply]
If any updates, could you add link to this discussion, thanks~!
